From 08ce2fc55a0296d5cf3f93defcc2ee01181aa3dd Mon Sep 17 00:00:00 2001
From: Mike Gerdts <mike.gerdts@joyent.com>
Date: Wed, 31 Jul 2019 18:30:14 +0000
Subject: [PATCH] OS-7916 bash completions for vmadm snapshot commands do not
 support bhyve Reviewed by: Patrick Mooney <patrick.mooney@joyent.com>
 Reviewed by: John Levon <john.levon@joyent.com> Approved by: John Levon
 <john.levon@joyent.com>

---
 src/vm/etc/vmadm.completion | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/src/vm/etc/vmadm.completion b/src/vm/etc/vmadm.completion
index 0e9fbc78..89871e13 100644
--- a/src/vm/etc/vmadm.completion
+++ b/src/vm/etc/vmadm.completion
@@ -41,9 +41,8 @@ _vmadm()
         || ${prev} == 'delete-snapshot'
         || ${prev} == 'rollback-snapshot' ]]; then
 
-        # OS only commands
-        vms_uuids=$(zoneadm list -cp | grep -v ':global:' \
-            | grep -v ":[bh]*[ky]v[em]:excl:" | cut -d':' -f5 | sort | uniq)
+        vms_uuids=$(zoneadm list -cp |
+	    nawk -F: '$2 != "global" && $6 != "kvm" { print $5 }' | sort -u)
         COMPREPLY=( $(compgen -W "${vms_uuids}" -- ${cur}) )
 
     elif [[ ${prev} == 'validate' ]]; then
-- 
2.21.0

